lib/src/components/NxFileUpload/NxMultiFileUpload/NxMultiFileUpload.tsx

```diff
@@ -0,0 +1,180 @@
+        'nx-multi-file-upload__no-file-message--invalid': showError
+      }),
+      inputRef = useRef<HTMLInputElement>(null),
+      inputId = useUniqueId('nx-multi-file-upload-input', id),
+      selectedFilesContainerRef = useRef<HTMLDivElement>(null),
+      validationErrorId = useUniqueId('nx-multi-file-upload-validation-error');
+
+  function onChange(evt: FormEvent<HTMLInputElement>) {
+    const dataTransferObject = new DataTransfer();
+
+    // get the previous files uploaded
+    if (files) {
+      forEach((f) => dataTransferObject.items.add(f), Array.from(files));
+    }
+
+    const inputFiles = evt.currentTarget.files,
+        normalizedFiles = !inputFiles?.length ? null : inputFiles;
+
+    if (normalizedFiles) {
+      forEach((f) => dataTransferObject.items.add(f), Array.from(normalizedFiles));
+    }
+
+    onChangeProp(dataTransferObject.files);
+  }
+
+  function openPicker() {
+    inputRef.current?.click();
+  }
+
+  function onDismiss(fileObj: File) {
+    const closeBtns = Array.from(selectedFilesContainerRef.current?.
+        querySelectorAll<HTMLButtonElement>('.nx-selected-file__dismiss-btn') ?? []);
+
+    if (files) {
+      const filesArray = Array.from(files);
+      filesArray.map((f) => {
+        if (f === fileObj) {
+          const idx = filesArray.indexOf(f);
+          if (idx === files.length - 1) {
+            // if the selected file is the last in the container, set the focus to the next last dismiss button
+            // if selected file is only file, set focus on the input
+            closeBtns.length === 1 ? inputRef.current?.focus() : closeBtns[idx - 1].focus();
+          }
+          else {
+            // need to explicity set focus on next closeBtn for Safari
+            closeBtns[idx].focus();
+          }
+        }
+      });
+    }
+
+    const dataTransferObject = new DataTransfer();
+    if (inputRef.current) {
+      if (files) {
+        forEach((f: File) => dataTransferObject.items.add(f), without([fileObj], Array.from(files)));
+
+        const normalizedFiles = !dataTransferObject.files.length ? null : dataTransferObject.files;
+        onChangeProp(normalizedFiles);
+      }
+    }
+  }
+
+  useEffect(function() {
+    if (inputRef.current) {
+      if (isFileSelected) {
+        inputRef.current.files = files;
+      }
+      else {
+        // there's no way to clear the list via the files prop, but this does it
+        inputRef.current.value = '';
+      }
+    }
+  }, [files]);
+
+  return (
+    <div ref={ref} className={className} >
+      <div className="nx-multi-file-upload__container">
+        <input ref={inputRef}
+               { ...attrs }
+               disabled={disabled}
+               onChange={onChange}
+               id={inputId}
+               className="nx-multi-file-upload__input"
+               type="file"
+               aria-label={isFileSelected ? `${files.length} files selected` : 'No File Selected'}
```

Comment:
The accessible name of the input needs to typically be the text of the `<label>` within which it is located. This `aria-label` is overriding that which is no good.  We should still use `aria-describedby` like `NxFileUpload` does.  I guess you changed it because the "x files selected" text isn't present anywhere in the DOM that you can reference? Just add an element that contains it, hidden with `display: none`.  I'm pretty certain that elements with `display: none` can still be used for accessible descriptions.  Double check of course, using the accessibility tab in the browser dev tools